server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -124,8 +128,16 @@ getExecPlanPartial userInfo sc req = do
     VT.RemoteType _ rsi ->
       return $ GExPRemote rsi opDef
   where
+    role = userRole userInfo
     gCtxRoleMap = scGCtxMap sc
 
+    checkQueryInAllowlist =
+      -- only for non-admin roles
+      when (role /= adminRole) $ do
+        let notInAllowlist =
+              not $ VQ.isQueryInAllowlist (_grQuery req) (scAllowlist sc)
+        when notInAllowlist $ throwVE "query is not in allowlist"
```

Comment:
The error message should say, "query is not allowed". An internal message can be `query is not in any of the allowlists` which will be logged on the server.